### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by Hashim-47**
Task 6: GET /api/articles/:article_id/comments

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -100,4 +100,49 @@ test("Respond with code 400 when article id does not exist", () => {
...
+        .then(({body}) => {
+            expect(body.comments.length).toBeGreaterThan(1);
+            body.comments.forEach((comment) => {
+                expect(comment).toHaveProperty("comment_id");
+                expect(comment).toHaveProperty("votes");
+                expect(comment).toHaveProperty("created_at");
+                expect(comment).toHaveProperty("author");
+                expect(comment).toHaveProperty("body");
+                expect(comment).toHaveProperty("votes");
+                expect(comment).toHaveProperty("article_id");
```
Comment (jimstev2447) [inline __tests__/app.test.js:127]: I know you are checking that it has an article_id property but should all the comments have an article_id of 1 in this case?

You could be more specific in this test :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -100,4 +100,49 @@ test("Respond with code 400 when article id does not exist", () => {
...
+
+test("Respond with code 400 when article id does not exist when looking for comments", () => {
+    return request(app).get("/api/articles/word/comments")
+    .expect(400)
+    .then(({body}) => {
+    expect(body.msg).toBe("Bad Request");
+})
+})
+
+test("Respond with code 200 and sort all comments by date in descending order", () => {
```
Comment (jimstev2447) [inline __tests__/app.test.js:149]: Have you checked to see if the server would respond correctly if a review currently has no comments (i.e it returns a 200 with an empty array)

**Review by jimstev2447 — COMMENTED** [review]
You need to add a test to make sure you get back an empty array an a 200 status for a review that doesnt have any comments, once you've done that and got all your other tests passing you can merge in

File: __tests__/app.test.js (hunk 2 above)
Comment (Rosem10) [reply __tests__/app.test.js:149]: This still needs to be actioned, please! It's good to check that the array has more than one value before checking each comment's keys, but please also add a test for if we receive an empty array back - nothing has gone wrong - there just aren't any associated comments

**Review by Rosem10 — APPROVED** [review]
Please add in that check we've mentioned before merging!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -100,4 +100,59 @@ test("Respond with code 400 when article id does not exist", () => {
 })
 })
 })
-})
\ No newline at end of file
+})
+
+describe('GET /api/articles/:id/comments', () =>{
+
+    test('Responds with code 200 and an empty array when passed an article with no comments', () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:107]: Ah I see now!

**Review by Rosem10 — APPROVED** [review]
